### PR TITLE
Support ES Modules.

### PR DESCRIPTION
**Pull request by threepointone**
This PR provides a possible solution for getting Workers Sites working with ES Module workers. This approach is not as invasive as other approaches, so isn't as risky either.

Usage:

```jsx
import manifestJSON from "__STATIC_CONTENT_MANIFEST";
const manifest = JSON.parse(manifestJSON);

export default {
  fetch(request, env, ctx) {
    return await getAssetFromKV(
      {
        request,
        waitUntil(promise) {
          return ctx.waitUntil(promise);
        },
      },
      {
        ASSET_NAMESPACE: env.ASSET_NAMESPACE,
        ASSET_MANIFEST: manifest,
      }
    );
    // ...
  },
};
```

I'll also send a PR to wrangler that adds the manifest as a text module `__STATIC_CONTENT_MANIFEST` to the worker.

cc @xortive Is this approach right? It's following your comment here https://github.com/cloudflare/wrangler/issues/1938#issuecomment-867200289

### CODE REVIEW

**Comment by threepointone** [comment]
You don't even have to merge this PR really, just to be more accurate about the type signature

File: src/index.ts [hunk 1]
```diff
@@ -103,7 +103,13 @@ function serveSinglePageApp(request: Request, options?: Partial<Options>): Reque
  * @param {Object | string} [options.ASSET_NAMESPACE] the binding to the namespace that script references
  * @param {any} [options.ASSET_MANIFEST] the map of the key to cache and store in KV
  * */
-const getAssetFromKV = async (event: FetchEvent, options?: Partial<Options>): Promise<Response> => {
+
+type Evt = {
+  request: Request
+  waitUntil: (promise: Promise<any>) => void
+}
```
Comment (SukkaW) [inline src/index.ts:110]: Why custom `Evt` type? You know there are [`@cloudflare/workers-types`](https://github.com/cloudflare/workers-types) out there, right?

**Review by SukkaW — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (threepointone) [reply src/index.ts:110]: I chose the word "Evt" to be familiar, and `@cloudflare/workers-types` won't ever define the types for kv-asset-handler.

**Review by Cherry — APPROVED** [review]
Makes sense to me, thanks!

**Pull request merged**